packages/cli/commands/project/deploy.js

```diff
@@ -30,6 +30,26 @@ const { uiCommandReference, uiAccountDescription } = require('../../lib/ui');
 exports.command = 'deploy [--project] [--buildId]';
 exports.describe = uiBetaTag(i18n(`${i18nKey}.describe`), false);
 
+const validateBuildId = (
+  buildId,
+  deployedBuildId,
+  latestBuildId,
+  projectName
+) => {
+  if (Number(buildId) > latestBuildId) {
+    return i18n(`${i18nKey}.errors.buildIdDoesNotExist`, {
+      buildId: buildId,
+      projectName,
+    });
+  }
+  if (Number(buildId) === deployedBuildId) {
+    return i18n(`${i18nKey}.errors.buildAlreadyDeployed`, {
+      buildId: buildId,
+    });
+  }
+  return true;
+};
+
```

Comment:
Non blocking but could see this making sense living in `lib/projects`

Reply:
Yeah, I was wondering about that myself.  I can do it as a follow up PR if that works